src/pages/resultsView/coExpression/CoExpressionTabUtils.ts

```diff
@@ -28,17 +28,60 @@ export function filterAndSortProfiles(profiles:MolecularProfile[]) {
         }
         return good;
     });
-    profs.sort(function(profA, profB) {
-        // sort rna seq to the top
-        const rnaSeqA = profA.molecularProfileId.toLowerCase().indexOf("rna_seq") > -1;
-        const rnaSeqB = profB.molecularProfileId.toLowerCase().indexOf("rna_seq") > -1;
-        if (rnaSeqA === rnaSeqB) {
-            return 0;
-        } else if (rnaSeqA) {
-            return -1;
-        } else {
-            return 1;
+    return profs;
+}
+
+export function getGenesetProfiles(profiles:MolecularProfile[]) {
+    const profs = profiles.filter(profile=>{
+        // we want a profile which is mrna or protein, and among them we want any profile,
+        // except zscore profiles that are not merged_median_zscores
+
+        let good = false;
+        if (profile.molecularAlterationType === AlterationTypeConstants.GENESET_SCORE) {
+
+            const profileId = profile.molecularProfileId.toLowerCase();
+            good = profileId.indexOf("gsva_pvalues") === -1;
         }
+        return good;
     });
     return profs;
+}
+
+export function sortProfiles(profs: MolecularProfile[]) {
+    // sort rna seq to the top
+    return _.sortBy(profs, profile=>(profile.molecularProfileId.toLowerCase().indexOf("rna_seq") > -1) ? 0 : 1);
+}
+
+export function filterAndSortProfilesY(profiles:MolecularProfile[]) {
+    const allProfs = getGeneProfiles(profiles).concat(getGenesetProfiles(profiles));
+    return sortProfiles(allProfs);
+}
+
+export function filterAndSortProfilesX(geneticEntity:"gene"|"geneset", profiles:MolecularProfile[]) {
```

Comment:
for example,`GeneticEntityType` enum should be type of `geneticEntity` parameter (probably rename the parameter to `geneticEntityType` too